### DISCUSSION TITLE
I don't understand the behavior of the scheduler synchronization

### DISCUSSION THREAD

**Discussion by xaviermarchegay** [Guikingone/SchedulerBundle#216] [Q&A] [answered]
The scheduler randomly crashes with the following error. It breaks about one of twenty times so I don't really care but I would like to understand. :)

`[2021-12-17T16:14:15.514281+00:00] console.CRITICAL: Error thrown while running command "'scheduler:consume'". Message: "The scheduler is not synchronized with the current clock, current delay: 0 microseconds, allowed range: [1000000, 86400000000]" {"exception":"[object] (SchedulerBundle\\Exception\\RuntimeException(code: 0): The scheduler is not synchronized with the current clock, current delay: 0 microseconds, allowed range: [1000000, 86400000000] at /var/www/html/vendor/guikingone/scheduler-bundle/src/Scheduler.php:294)","command":"'scheduler:consume'","message":"The scheduler is not synchronized with the current clock, current delay: 0 microseconds, allowed range: [1000000, 86400000000]"} []`

I launch the scheduler every minute in my host crontab : 

`/usr/bin/docker exec -u www-data picophony-php /bin/bash -c "bin/console scheduler:consume"`

Thanks for this package

**Comment by Guikingone** [comment, marked as answer]
Hi @xaviermarchegay 👋🏻 

The goal of the synchronisation is to ensure that the scheduler is "up to date" regarding the VM / OS / Container internal clock. 

Regarding the error, I might know why it happen, due to PHP 8.0 migration, the internal logic used to check this synchronisation has been changed: https://github.com/Guikingone/SchedulerBundle/blob/main/src/Scheduler.php#L289, it might explain why you're encountering the error. 

Just to be sure, when you're launching: 

```bash
/usr/bin/docker exec -u www-data picophony-php /bin/bash -c "bin/console scheduler:consume"
```

It is launched using cron via a container or directly in the VM / server?

Thanks for the precision and sorry for this bug, I'm gonna fix it ASAP 🙂

**Comment by xaviermarchegay** [comment]
The command is set in the crontab of the server.

Again, it is not a big deal since it almost works all the time and it is a personal project.
